app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/CreateAccountActivityTest.java

```diff
@@ -0,0 +1,531 @@
+
+        solo.enterText(emailText, resources.getString(R.string.test_create_account_email));
+        solo.enterText(usernameText, resources.getString(R.string.test_create_account_username));
+        solo.enterText(phoneNumberText, resources.getString(R.string.test_account1_phone));
+
+        solo.clickOnButton(resources.getString(R.string.create_account));
+
+        Assert.assertTrue(solo.waitForText(EditTextValidator.WEAKPASS));
+
+        Assert.assertNotNull(password1Text.getError());
+        Assert.assertNotNull(password2Text.getError());
+
+        Assert.assertNull(usernameText.getError());
+        Assert.assertNull(emailText.getError());
+        Assert.assertNull(phoneNumberText.getError());
+    }
+
+    /**
+     * Tests create account button with password 5 characters and a space
+     */
+    @Test public void shortPasswordAndSpaceTest()
+    {
+        solo.enterText(password1Text, resources.getString(R.string.short_pass) + resources.getString(R.string.space));
+        solo.enterText(password2Text, resources.getString(R.string.short_pass) + resources.getString(R.string.space));
+
+        solo.enterText(emailText, resources.getString(R.string.test_create_account_email));
+        solo.enterText(usernameText, resources.getString(R.string.test_create_account_username));
+        solo.enterText(phoneNumberText, resources.getString(R.string.test_account1_phone));
+
+        solo.clickOnButton(resources.getString(R.string.create_account));
+
+        Assert.assertTrue(solo.waitForText(EditTextValidator.WEAKPASS));
+
+        Assert.assertNotNull(password1Text.getError());
+        Assert.assertNotNull(password2Text.getError());
+
+        Assert.assertNull(usernameText.getError());
+        Assert.assertNull(emailText.getError());
+        Assert.assertNull(phoneNumberText.getError());
+    }
+
+    /**
+     * Tests successful account creation and signs out
+     */
+    @Test
+    public void createAccountTest()
+    {
+        solo.enterText(emailText, resources.getString(R.string.test_create_account_email));
+        solo.enterText(usernameText, resources.getString(R.string.test_create_account_username));
+
+        solo.enterText(phoneNumberText, resources.getString(R.string.test_account1_phone));
+        solo.enterText(password1Text, resources.getString(R.string.test_account1_password));
+        solo.enterText(password2Text, resources.getString(R.string.test_account1_password));
+
+        solo.clickOnButton(resources.getString(R.string.create_account));
+
+        solo.assertCurrentActivity(resources.getString(R.string.wrong_activity), MyBooksActivity.class);
+
+        createAccountSignOut();
+    }
+
+    /**
+     * Signs out of created test account after deleting account
+     */
+    public void createAccountSignOut()
+    {
+        solo.clickOnText(resources.getString(R.string.navbar_text_label_4));
+
+        solo.waitForText(resources.getString(R.string.sign_out));
```

Comment:
why does it click on the sign_out button twice?

Reply:
oh, thats after loading Profile wait for page to load before deleting account

Reply:
here so its more readable :p

![image](https://user-images.githubusercontent.com/39696855/98030278-076d3c80-1dce-11eb-9468-13278bac8ec6.png)

Reply:
ohhhh I see. I think it was fine how it was before, just in case in the future we show the username somewhere.